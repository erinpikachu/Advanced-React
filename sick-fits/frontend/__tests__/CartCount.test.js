import { render } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Martches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('Updates via props', async () => {
    const { container, rerender } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');
    // Update the props
    rerender(<CartCount count={12} />);
    // wait for __ ms
    expect(container.textContent).toBe('1211');
    await wait(400);
    // await screen.findByAltText('12');
    expect(container.textContent).toBe('12');
    expect(container).toMatchSnapshot();
  });
});
