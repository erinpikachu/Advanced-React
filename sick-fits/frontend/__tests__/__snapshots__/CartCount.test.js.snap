// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Martches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 eILtgZ"
  >
    <div>
      <div
        class="CartCount__DotStyles-sc-xxvp4g-0 ggjQds count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 eILtgZ"
  >
    <div>
      <div
        class="CartCount__DotStyles-sc-xxvp4g-0 ggjQds count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
